pkg/sfu/streamtracker.go

```diff
@@ -27,16 +28,20 @@ const (
 // It runs its own goroutine for detection, and fires OnStatusChanged callback
 type StreamTracker struct {
 	// number of samples needed per cycle
-	SamplesRequired uint32
+	samplesRequired uint32
 	// number of cycles needed to be active
-	CyclesRequired  uint64
-	CycleDuration   time.Duration
-	OnStatusChanged func(StreamStatus)
-	initialized     atomicBool
-	paused          atomicBool
-	status          atomicInt32 // stores StreamStatus
-	countSinceLast  uint32      // number of packets received since last check
-	running         chan struct{}
+	cyclesRequired uint64
+	cycleDuration  time.Duration
+
+	onStatusChanged func(status StreamStatus)
+
+	initialized    atomicBool
+	paused         atomicBool
+	countSinceLast uint32 // number of packets received since last check
+	running        chan struct{}
+
+	statusMu sync.RWMutex
```

Comment:
@davidzhao Introducing this lock as I think I was hitting the two goroutines problem you told me about where each was having some thread local stuff. I am not really sure though as I was getting layer stops randomly. As this is a low frequency usage, made it a lock.

Reply:
good call. you are right there could be a race. I wanted to use the atomic swap method but it was in a different package (atomicFlag). mutex works though!